hilti/runtime/include/types/map.h

```diff
@@ -222,6 +222,19 @@ class Map : protected std::map<K, V> {
         }
     }
 
+    /**
+     * Attempts to get the value for a key.
+     *
+     * @param k key to retrieve
+     * @return the value, or an unset optional if the key is not set in the map
+     */
+    std::optional<V> get_optional(const K& k) const& {
```

Comment:
Just wondering about the name, is maybe `try_get` more intuitive than `get_optional`? No strong preference though.

Reply:
Actually I had considered `try_get` and also `get_try`, but ended up preferring `get_optional`. I'll leave it at that.